astropy/modeling/fitting.py

```diff
@@ -549,24 +549,32 @@ def __call__(self, model, x, y, z=None, weights=None, **kwargs):
         fitted_model = self.fitter(model, x, y, z, weights=weights, **kwargs)
         if z is None:
             filtered_data = y
+            filtered_weights = weights
```

Comment:
Although it fits logically with the surrounding code, I might move this line out of the condition, to avoid replication.

Reply:
ok